lib/ansible/modules/storage/zfs/zpool.py

```diff
@@ -0,0 +1,454 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: (c) 2018, Remy Mudingay <remy.mudingay[at]esss.se>
+# Copyright: (c) 2018, Stephane Armanet <stephane.armanet[at]esss.se>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: zpool
+short_description: Manage zfs zpools
+version_added: "2.7"
+description:
+  - Manage virtual storage pools using zfs zpools
+options:
+  name:
+    description:
+      - Pool name
+    required: true
+  add:
+    description:
+      - Add devices (spare or mirror to an existing zpool)
+    type: bool
+    default: false
+  raid_level:
+    description:
+      - Pool raid level
+    choices: [ raid0, mirror, raidz, raidz1, raidz2, raidz3 ]
+  vdev:
```

Comment:
I would rename this to vdevs and add "vdev" as alias.

Reply:
done.